packages/app-store/office365calendar/lib/CalendarService.ts

```diff
@@ -454,12 +454,13 @@ export default class Office365CalendarService implements Calendar {
       (acc: BufferedBusyTime[], subResponse: { body: { value?: BodyValue[]; error?: Error[] } }) => {
         if (!subResponse.body?.value) return acc;
         return acc.concat(
-          subResponse.body.value
-            .filter((evt) => evt.showAs !== "free" && evt.showAs !== "workingElsewhere")
-            .map((evt) => ({
```

Comment:
If we chain `.filter` and `.map` we are essentially iterating through the array twice. By replacing this with a `.reduce` we're able to filter through values while only iterating through the array once.

Reply:
`filter` and `map` usage keeps the code clean and simple to understand. Unless the array is huge it doesn't really matter performance wise how many times we iterate over it.

What's the performance improvement in here?

Reply:
lets do this in a follow up PR

Reply:
@hariombalhara there was a user that was experiencing a long load time for availability from their Outlook calendar. From what I gathered the payload was very large. I think most of the time the busy times array is small enough that it's fine to do the `filter` and `map` but in this scenario, we're running into the theoretical O(2n) vs O(n).

Reply:
Thanks. Makes sense then.